lib/Interpreter/Value.cpp

```diff
@@ -231,9 +231,15 @@ namespace cling {
   namespace valuePrinterInternal {
     std::string printTypeInternal(const Value& V);
     std::string printValueInternal(const Value& V);
+    extern const char* const kUndefined;
   } // end namespace valuePrinterInternal
 
   void Value::print(llvm::raw_ostream& Out, bool Escape) const {
+    if (!m_Interpreter) {
+      Out << valuePrinterInternal::kUndefined << "\n";
```

Comment:
Because the error is already reported, and what "invalid" means is really not clear.  How is it invalid? What aspect.
Undefined represents a value in many context's whose content should not be relied upon to consistently be what was shown.

It's not test simplicity, it using and parsing the output (especially in a shell needs to trap two or three distinct cases that could grow even more as people invent new problems to represent when they should be reporting it properly):

VAL=$(cling blah blah)
```
if [[ $VAL != *"<undefined>"* ]]; then
 ...
fi
```
vs.

```
if [[ $VAL != *"<<<invalid>>>"* ]] || [[ $VAL != *"ERROR in cling"* ]]
 ...
fi
```

And the above example doesn't even take into account the bug that was fixed which would then have a totally new string representation, or if it didn't would suffer the same problem of Value.cpp needing a declaration to share.